StardewArchipelago/GameModifications/Modded/JunimoShopGenerator.cs

```diff
@@ -0,0 +1,504 @@
+        private static readonly List<string> RedColors = new()
+        {
+            "color_red", "color_pink", "color_dark_pink", "color_salmon"
+        };
+        private Dictionary<int, int> YellowItems { get; set; }
+        private static readonly List<string> YellowColors = new()
+        {
+            "color_yellow", "color_gold", "color_sand", "color_dark_yellow"
+        };
+        private Dictionary<int, int> OrangeItems { get; set; }
+
+        private static readonly List<string> OrangeColors = new()
+        {
+            "color_orange", "color_dark_orange", "color_dark_brown", "color_brown", "color_copper"
+        };
+        public Dictionary<int, int> PurpleItems {get; set; }
+        private static readonly List<string> PurpleColors = new()
+        {
+            "color_purple", "color_dark_purple", "color_dark_pink", "color_pale_violet_red", "color_iridium"
+        };
+        public Dictionary<StardewItem, int> BerryItems {get; set; }
+        private static readonly string[] spring = new string[]{"spring"};
+        private static readonly string[] summer = new string[]{"summer"};
+        private static readonly string[] fall = new string[]{"fall"};
+
+        private class JunimoVendor
+        {
+            public PersistentStock JunimoPersistentStock {get; private set;}
+            public Dictionary<int, int> ColorItems {get; private set;}
+
+            public JunimoVendor(PersistentStock junimoPersistentStock, Dictionary<int, int> colorItems)
+            {
+                JunimoPersistentStock = junimoPersistentStock;
+                ColorItems = colorItems;
+            }
+        }
+
+        public JunimoShopGenerator(
+            ArchipelagoClient archipelago, ShopStockGenerator shopStockGenerator, StardewItemManager stardewItemManager)
+        {
+            _archipelago = archipelago;
+            _shopStockGenerator = shopStockGenerator;
+            _stardewItemManager = stardewItemManager;
+            RedPersistentStock = new PersistentStock();
+            GreyPersistentStock = new PersistentStock();
+            BluePersistentStock = new PersistentStock();
+            YellowPersistentStock = new PersistentStock();
+            OrangePersistentStock = new PersistentStock();
+            GenerateItems();
+            var blueVendor = new JunimoVendor(BluePersistentStock, BlueItems);
+            var yellowVendor = new JunimoVendor(YellowPersistentStock, YellowItems);
+            var greyVendor = new JunimoVendor(GreyPersistentStock, GreyItems);
+            var redVendor = new JunimoVendor(RedPersistentStock, RedItems);
+            var orangeVendor = new JunimoVendor(OrangePersistentStock, OrangeItems);
+            JunimoVendors = new Dictionary<string, JunimoVendor>(){
+                {"Blue", blueVendor}, {"Yellow", yellowVendor}, {"Grey", greyVendor}, {"Red", redVendor}, {"Orange", orangeVendor}, 
+            };
+        }
+
+        public void GenerateItems()
```

Comment:
This method, and to a lesser degree a lot of the methods below it, have a truckload of inlined strings that could be constants